### PR TITLE
Add BlockFormEvent when a obsidian or cobblestone is generated from liquid

### PR DESCRIPTION
**Pull request by MrPowerGamerBR**
Something @imjack wanted and I thought it was a good idea to implement to Nukkit's API.

**Note:** Bukkit doesn't use BlockFormEvent for Obsidian/Cobblestone generated from liquids, however Nukkit doesn't have a BlockFromToEvent, so I used BlockFormEvent, if using BlockFromToEvent is better, then ask that I will change the code.

### CODE REVIEW

**Comment by NycuRO** [comment]
This fix bug visual of creating cobblestone. 👍 Tested. @Snake1999 You can accept?

**Comment by MagicDroidX** [comment]
I wonder how it is in Bukkit? And maybe we can improve the BlockUpdateEvent i guess?

**Comment by NycuRO** [comment]
@MagicDroidX Creeper have files of Spigot here. https://github.com/Creeperface01/MinecraftSource

**Comment by MrPowerGamerBR** [comment]
@NycuRO I'm not sure why it would fix that

@MagicDroidX in Bukkit it is BlockFromToEvent (from water to
obsidian/cobble)

Em 10 de out de 2016 8:29 AM, "MagicDroidX" <notifications@github.com>
escreveu:

**Comment by NycuRO** [comment]
@MrPowerGamerBR This is not implemented. Cobblestone get visual bug if isn't implemented. With this commit, you fixed bug . Thanks :)

**Comment by MagicDroidX** [comment]
I appreciate your idea but I think maybe `BlockFromToEvent` is better. I will merge it if you would like to change it. 👍

**Comment by NycuRO** [comment]
@MagicDroidX Magic,but need fix buggy. Is correct and not affected performance...

File: src/main/java/cn/nukkit/block/BlockLiquid.java [hunk 1]
```diff
@@ -457,9 +458,22 @@ private void checkForHarden() {
...
+                    newState.setComponents(this.getX(), this.getY(), this.getZ());
+                    newState.setLevel(this.getLevel());
+                    BlockFormEvent ev = new BlockFormEvent(this, newState);
+                    this.getLevel().getServer().getPluginManager().callEvent(ev);
+                    if (!ev.isCancelled()) {
+                        this.getLevel().setBlock(this, newState, true);
+                    }
                 } else if (this.getDamage() <= 4) {
-                    this.getLevel().setBlock(this, new BlockCobblestone(), true);
+                    Block newState = new BlockObsidian();
```
Comment (MagicDroidX) [inline src/main/java/cn/nukkit/block/BlockLiquid.java]: Block newState = new BlockCobblestone();

Comment (NycuRO) [inline src/main/java/cn/nukkit/block/BlockLiquid.java]: @MrPowerGamerBR

**Comment by MrPowerGamerBR** [comment]
@MagicDroidX will change it to BlockFromToEvent as soon as possible 👍

**Pull request closed without merging**